src/effects/RealtimeEffectState.cpp

```diff
@@ -69,9 +69,14 @@ class RealtimeEffectState::AccessState : public NonInterferingBase {
       mChannelFromMain.Read<FromMainSlot::Reader>(mEffect, mState);
    }
    void WorkerWrite() {
+
+      std::unique_lock lk(mLockForCV);
+
       // Worker thread avoids memory allocation.
       mChannelToMain.Write(CounterAndOutputs{
          mState.mWorkerSettings.counter, mState.mOutputs.get() });
+
+      mCV.notify_one();
```

Comment:
As it says here https://en.cppreference.com/w/cpp/thread/condition_variable/notify_one, it is not necessary to keep the mutex locked while signalling, and might even be worse for performancde.  It needs to be held for the duration of changes that make the condition the waiter waits on true.

So you might insert `{` before the declaration of `lk` and `}` before you call `noitfy_one`.

Reply:
However, I based my choice on what this page https://embeddedartistry.com/blog/2022/01/10/remember-to-lock-around-all-stdcondition_variable-variables/ says - search for this text: "However, it seems that the [C](https://embeddedartistry.com/fieldmanual-terms/c/)++ Core Guidelines discussions on GitHub prefer [this general guideline](https://github.com/isocpp/CppCoreGuidelines/issues/925#issuecomment-302269850):"

Reply:
but of course I will give a try to your suggestion